lib/Frontend/Frontend.cpp

```diff
@@ -343,7 +343,8 @@ void CompilerInstance::setupDependencyTrackerIfNeeded() {
     // If we have an output path specified, but no other tracking options,
     // default to non-system dependency tracking.
     if (opts.InputsAndOutputs.hasDependencyTrackerPath() ||
-        !opts.IndexStorePath.empty()) {
+        !opts.IndexStorePath.empty() ||
+        opts.RequestedAction == FrontendOptions::ActionType::EmitModuleOnly) {
```

Comment:
@CodaFi What do you think of forcing dependency tracking for all emit module only jobs? The data structure configured here is used later to generate the fine grained dependency graph which is also now built by default with all emit module only jobs.

Reply:
This part of the code is about forcing system (read: SDK) dependencies on and off when we import stuff via the clang importer. I don't think you need to amend this logic. The user should be the one in control here with -track-system-dependencies.

Reply:
This change is no more necessary when fixing the SourceFile issue above.